src/lang-promql/parser/parser.ts

```diff
@@ -269,31 +270,32 @@ export class Parser {
       retrieveAllRecursiveNodes(walkThrough(node, LabelMatchers, LabelMatchList), LabelMatchList, LabelMatcher),
       this.state
     );
-    // A Vector selector must contain at least one non-empty matcher to prevent
-    // implicit selection of all metrics (e.g. by a typo).
-    let empty = true;
-    let i = 0;
-    while (i < labelMatchers.length && empty) {
-      empty = labelMatchers[i].matchesEmpty();
-      i++;
-    }
-    if (labelMatchers.length > 0 && empty) {
-      this.addDiagnostic(node, 'vector selector must contain at least one non-empty matcher');
-    }
     let vectorSelectorName = '';
     // VectorSelector ( MetricIdentifier ( Identifier ) )
     // https://github.com/promlabs/lezer-promql/blob/71e2f9fa5ae6f5c5547d5738966cd2512e6b99a8/src/promql.grammar#L200
     const vectorSelectorNodeName = node.firstChild?.firstChild;
     if (vectorSelectorNodeName) {
       vectorSelectorName = this.state.sliceDoc(vectorSelectorNodeName.start, vectorSelectorNodeName.end);
     }
+    if (vectorSelectorName !== '') {
+      labelMatchers.push(new Matcher(EqlSingle, '__name__', vectorSelectorName));
+    }
+    console.log(`${this.tree}`);
```

Comment:
You forgot some debug statements :)

Reply:
mmm I should enable the rule in eslint to stop forgetting them